unturned/OpenMod.Unturned/Users/UnturnedUserProvider.cs

```diff
@@ -340,6 +340,61 @@ async UniTask BroadcastTask()
 
             return BroadcastTask().AsTask();
         }
+        
+        public async Task<bool> BanAsync(IUser user, string? reason = null, DateTime? endTime = null)
+        {
+            var player = m_Users.FirstOrDefault(x => x.Id == user.Id);
+
+            if(player == null)
+            {
+                return false;
+            }
+
+            if (reason == null)
+            {
+                reason = "No reason provided";
+            }
+
+            if(endTime == null)
+            {
+                endTime = DateTime.Now;
+            }
+
+            async UniTask BanAsync()
+            {
+                await UniTask.SwitchToMainThread();
+                Provider.ban(player.SteamId, reason, (uint)(endTime.Value - DateTime.Now).TotalSeconds);
```

Comment:
This will throw a NullReferenceException when endTime is null, add this before it:

```cs
endTime ??= DateTime.MaxValue
```